Northstar.Custom/mod/scripts/vscripts/sh_server_dialog.gnut

```diff
@@ -0,0 +1,259 @@
+
+	foreach( string arg in args )
+	{
+		command = format( "%s %s", command, arg )
+	}
+	
+	ServerToClientStringCommand( player, command )
+}
+
+void function SendMessageServerDialog( entity player, string message )
+{
+	ServerDialogData dialog
+	dialog.message = message
+	SendServerDialog( player, dialog )
+}
+
+void function AddDialogButton( ServerDialogData dialog, string text , void functionref(entity) callback )
+{
+	ServerDialogButtonData button
+	button.text = text
+	button.callback = callback
+	dialog.buttonData.append( button )
+}
+
+bool function ClientCommand_Button_Clicked( entity player, array<string> args )
+{
+	if ( !IsValid( player ) || args.len() != 1 || args[0].tointeger() < 0 || args[0].tointeger() > 3 )
+		return true
+	
+	int index = args[0].tointeger()
+	
+	if ( !( player in file.playerButtonCallBack ) || file.playerButtonCallBack[player].len() < index + 1  )
+		return true
+
+	file.playerButtonCallBack[ player ][ index ]( player )
+
+	return true
+}
+
+#elseif CLIENT
+
+void function ServerDialogueOnClientInit()
+{
+    AddServerToClientStringCommandCallback( "ServerRequestResquest", ServerCallback_DialogueRequestReceived )
+}
+
+// 	string header
+// 	string message
+// 	string image = ""
+// 	string rightImage = ""
+// 	bool forceChoice = false
+// 	bool noChoice = false
+// 	bool noChoiceWithNavigateBack = false
+// 	bool showSpinner = false
+// 	bool showPCBackButton = false
+// 	float inputDisableTime = 0
+// 	bool darkenBackground = false
+// 	bool useFullMessageHeight = false
+
+// 	array<ServerDialogButtonData> buttonData
+// the long array is in this order
+
+void function ServerCallback_DialogueRequestReceived( array<string> args )
+{
+    if ( args.len() != 16 )
+        return
+        
+    string header = StringReplace( args[0], "./s.", " ", true, true )
+    string message = StringReplace( args[1], "./s.", " ", true, true )
+    string image = args[2]
+    string rightImage = args[3]
+    bool forceChoice = args[4].tointeger() == 1
+    bool noChoice = args[5].tointeger()  == 1
+    bool noChoiceWithNavigateBack = args[6].tointeger() == 1
+    bool showSpinner = args[7].tointeger() == 1
+    bool showPCBackButton = args[8].tointeger() == 1
+    float inputDisableTime = args[9].tofloat()
+    bool darkenBackground = args[10].tointeger() == 1
+    bool useFullMessageHeight = args[11].tointeger() == 1
+
+    string button0Text = StringReplace( args[12], "./s.", " ", true, true )
+    string button1Text = StringReplace( args[13], "./s.", " ", true, true )
+    string button2Text = StringReplace( args[14], "./s.", " ", true, true )
+    string button3Text = StringReplace( args[15], "./s.", " ", true, true )
+
+    RunUIScript( "ClientCallBack_RequestDialog", header, message, image, rightImage, forceChoice, noChoice, noChoiceWithNavigateBack, showSpinner, showPCBackButton, inputDisableTime, darkenBackground, useFullMessageHeight, button0Text, button1Text, button2Text, button3Text )
```

Comment:
Might be better to split this into multiple lines, for readability, although I'm not sure if squirrel likes this